src/lib/trajectory/parse.ts

```
@@ -936,46 +1035,74 @@ export function parse_pymatgen_trajectory(
 
   const lattice_params = math.calc_lattice_params(matrix)
 
   const frames: TrajectoryFrame[] = coords.map((frame_coords, frame_idx) => {
+    // Extract frame properties first to check for forces
+    const frame_props = frame_properties?.[frame_idx] || {}
+
+    // Extract forces if available (check multiple possible locations)
+    let forces_data: number[][] | undefined
+
+    // Method 1: frame_properties.forces.data (standard format)
+    if (frame_props.forces && typeof frame_props.forces === `object`) {
```

Biome (1.9.4)

[error] Invalid `typeof` comparison value: this expression is not a string literal

not a string literal

(lint/suspicious/useValidTypeof)